Kickstarter-iOS/Features/PledgedProjectsOverview/CardView/PPOAlertFlag.swift

```diff
@@ -0,0 +1,119 @@
+import Library
+import SwiftUI
+
+struct PPOAlertFlag: View {
+  let alert: Alert
+
+  var body: some View {
+    HStack {
+      self.image
+        .renderingMode(.template)
+        .aspectRatio(contentMode: .fit)
+        .foregroundStyle(self.foregroundColor)
+        .frame(width: Constants.imageSize, height: Constants.imageSize)
+      Spacer()
+        .frame(width: Constants.spacerWidth)
+      Text(self.alert.message)
+        .font(Font(Constants.font))
+        .foregroundStyle(self.foregroundColor)
+    }
+    .padding(Constants.padding)
+    .background(self.backgroundColor)
+    .clipShape(RoundedRectangle(cornerSize: CGSize(
+      width: Constants.cornerRadius,
+      height: Constants.cornerRadius
+    )))
+  }
+
+  var image: Image {
+    switch self.alert.type {
+    case .time:
+      Image(Constants.timeImage)
+    case .alert:
+      Image(Constants.alertImage)
+    }
+  }
+
+  var foregroundColor: Color {
+    switch self.alert.icon {
+    case .warning:
+      Color(uiColor: Constants.warningForegroundColor)
+    case .alert:
+      Color(uiColor: Constants.alertForegroundColor)
+    }
+  }
+
+  var backgroundColor: Color {
+    switch self.alert.icon {
+    case .warning:
+      Color(uiColor: Constants.warningBackgroundColor)
+    case .alert:
+      Color(uiColor: Constants.alertBackgroundColor)
+    }
+  }
+
+  struct Alert: Identifiable {
+    let type: AlertType
+    let icon: AlertIcon
+    let message: String
+
+    var id: String {
+      "\(self.type)-\(self.icon)-\(self.message)"
```

Comment:
The alert id can be same for same message, is this what we are tying to do here?

Reply:
Our schema doesn't have an identifier but we shouldn't get duplicates from the API anyway.